cmd/cluster-init/update_bootstrap_secrets.go

```diff
@@ -0,0 +1,342 @@
+	const chatBot = "ci-chat-bot"
+	name := chatBot + "-kubeconfigs"
+	keyAndField := serviceAccountKubeconfigPath(chatBot, o.clusterName)
+	return appendSecretItemContext(c, name, string(api.ClusterAPPCI), keyAndField, secretbootstrap.ItemContext{
+		Field: keyAndField,
+		Item:  chatBot,
+	})
+}
+
+func appendSecretItemContext(c *secretbootstrap.Config, name string, cluster string, key string, value secretbootstrap.ItemContext) error {
+	logrus.Infof("Appending secret item to: {name: %s, cluster: %s}", name, cluster)
+	sc, err := findSecretConfig(name, cluster, c.Secrets)
+	if err != nil {
+		return err
+	}
+	sc.From[key] = value
+	return nil
+}
+
+func updateExistingRegistryPullCredentialsAllSecrets(c *secretbootstrap.Config, o options) error {
+	for _, cluster := range c.UserSecretsTargetClusters {
+		if cluster != string(api.ClusterHive) && cluster != string(api.ClusterARM01) && cluster != o.clusterName {
+			return appendRegistrySecretItemContext(c, regPullCredsAll, cluster, secretbootstrap.DockerConfigJSONData{
+				AuthField:   fmt.Sprintf("token_image-puller_%s_reg_auth_value.txt", o.clusterName),
+				Item:        buildUFarm,
+				RegistryURL: registryUrlFor(o.clusterName),
+			})
+		}
+	}
+	return nil
+}
+
+func generateRegistryPullCredentialsAllSecrets(c *secretbootstrap.Config, o options) {
+	items := []secretbootstrap.DockerConfigJSONData{
+		{
+			AuthField:   "token_image-puller_ci_reg_auth_value.txt",
+			Item:        buildUFarm,
+			RegistryURL: api.ServiceAPPCIRegistry,
+		},
+		{
+			AuthField:   "auth",
+			Item:        "cloud.openshift.com-pull-secret",
+			RegistryURL: "cloud.openshift.com",
+			EmailField:  "email",
+		},
+		{
+			AuthField:   "auth",
+			Item:        "quay.io-pull-secret",
+			RegistryURL: "quay.io",
+			EmailField:  "email",
+		},
+		{
+			AuthField:   "auth",
+			Item:        "registry.connect.redhat.com-pull-secret",
+			RegistryURL: "registry.connect.redhat.com",
+			EmailField:  "email",
+		},
+		{
+			AuthField:   "auth",
+			Item:        "registry.redhat.io-pull-secret",
+			RegistryURL: "registry.redhat.io",
+			EmailField:  "email",
+		},
+	}
+	for _, cluster := range c.UserSecretsTargetClusters {
+		if cluster != string(api.ClusterHive) {
+			items = append(items, secretbootstrap.DockerConfigJSONData{
+				AuthField:   registryPullTokenField(cluster),
+				Item:        buildUFarm,
+				RegistryURL: registryUrlFor(cluster),
+			})
+		}
+	}
+	sc := secretbootstrap.SecretConfig{
+		From: map[string]secretbootstrap.ItemContext{
+			dotDockerConfigJson: generatePushPullSecretFrom(o.clusterName, items),
+		},
+		To: []secretbootstrap.SecretContext{
+			generateDockerConfigJsonSecretConfigTo(regPullCredsAll, ci, o.clusterName),
+			generateDockerConfigJsonSecretConfigTo(regPullCredsAll, testCredentials, o.clusterName),
+		},
+	}
+	logrus.Infof("Creating new secret with 'to' of: %v", sc.To)
+	c.Secrets = append(c.Secrets, sc)
+}
+
+func registryUrlFor(cluster string) string {
```

Comment:
nit: this is best placed in the same directory/file as all of the Cluster constants it references (and should take the cluster type as input, not a string)

Reply:
I could move this to that file, but I don't think using the cluster type will work as we will be dynamically adding new build clusters (such as `build03`) that will not pertain to a cluster type.